src/router/index.ts

```diff
@@ -1,10 +1,16 @@
 import { createRouter, createWebHistory } from "vue-router";
 
+const requiredAuthKeys = ["userId", "apiKey"];
+
 const routes = [
   {
     path: "/quiz/:quizId",
     name: "Player",
-    props: true,
+    props: (route: any) => ({
+      ...route.params,
+      userId: route.query.userId,
+      apiKey: route.query.apiKey,
+    }),
```

Comment:
why do you need this? We have been reading userId as a query param directly in `Player.vue` so far. Why can't we continue doing that?

Reply:
It was being used once before. Now that's it being used in multiple places, I think having a variable is better, no?

Reply:
It's fine by me. Will maybe wait for aman to respond